packages/dialog/src/react/index.js

```diff
@@ -87,32 +90,29 @@ class Dialog extends React.Component {
     this.handleKeyUp = this.handleKeyUp.bind(this)
   }
 
-  componentDidMount() {
-    if (this.el && !this.props.disableFocusOnMount) this.el.focus()
-  }
-
   handleKeyUp(evt) {
     if (evt.keyCode === ESCAPE_KEYCODE) this.props.onClose(evt)
   }
 
   render() {
     const { props } = this
+
+    const hasFocusLock = !!props.onClose
```

Comment:
should include `|| props.modal` ? (the case where we have a modal that doesn't feature a close button/callback)

Reply:
i will add that prop support